core/websock.js

```diff
@@ -45,58 +45,58 @@ export default class Websock {
     }
 
     // Getters and Setters
-    get_sQ() {
+    get sQ() {
         return this._sQ;
     }
 
-    get_rQ() {
+    get rQ() {
         return this._rQ;
     }
 
-    get_rQi() {
+    get rQi() {
         return this._rQi;
     }
 
-    set_rQi(val) {
+    set rQi(val) {
         this._rQi = val;
     }
 
     // Receive Queue
-    rQlen() {
+    get rQlen() {
         return this._rQlen - this._rQi;
     }
 
     rQpeek8() {
         return this._rQ[this._rQi];
     }
 
-    rQshift8() {
-        return this._rQ[this._rQi++];
-    }
-
-    rQskip8() {
-        this._rQi++;
+    rQskipBytes(bytes) {
+        this._rQi += bytes;
     }
 
-    rQskipBytes(num) {
-        this._rQi += num;
+    rQshift8() {
+        return this.rQshift(1);
     }
 
-    // TODO(directxman12): test performance with these vs a DataView
     rQshift16() {
-        return (this._rQ[this._rQi++] << 8) +
-               this._rQ[this._rQi++];
+        return this.rQshift(2);
     }
 
     rQshift32() {
-        return (this._rQ[this._rQi++] << 24) +
-               (this._rQ[this._rQi++] << 16) +
-               (this._rQ[this._rQi++] << 8) +
-               this._rQ[this._rQi++];
+        return this.rQshift(4);
+    }
+
+    // TODO(directxman12): test performance with these vs a DataView
+    rQshift(bytes) {
+        let res = 0;
+        for (let byte = bytes - 1; byte >= 0; byte--) {
+            res += this._rQ[this._rQi++] << (byte * 8);
```

Comment:
Have you done any performance tests? There is at least some risk that a for loop will be less efficient than the previous versions. And reading data can be done in very hot code paths.

Reply:
No I haven't.
There are definitely few extra operations. although, this could be optimized by the browser engine and, in my experience most of this micro-optimizations yield completely different results based on the browser engine.

Any hint on how would you like this to be tested?

Reply:
We have `tests/vnc_playback.html` that can be used together with the browsers profiler. I can have a look and see how it behaves.